docs/docs/adrs/adr-015-partial-set-security.md

```diff
@@ -0,0 +1,293 @@
+---
+sidebar_position: 16
+title: Partial Set Security
+---
+# ADR 015: Partial Set Security
+
+## Changelog
+* 2024-01-22: Proposed, first draft of ADR.
+
+## Status
+
+Proposed
+
+## Context
+Currently, in _Replicated Security_, the **whole** validator set of the Cosmos Hub _provider chain_ is used to secure consumer chains. There are at least three concerns with this approach. First, a big chunk of validators might be forced to validate consumer chains they are not interested in securing. Second, it is costly for small validators to secure additional chains. This concern is only partially addressed through [soft opt-out](https://github.com/cosmos/interchain-security/blob/main/docs/docs/adrs/adr-009-soft-opt-out.md) where the bottom (based on voting power) 5% validator can opt out from validating consumer chains. Third and for the above reasons, it is challenging for a new consumer chain to join Replicated Security.
+
+As a solution, we present _Partial Set Security_ (PSS). As the name suggests, PSS allows for only a subset of the validator set of Cosmos to secure a consumer chain. In what follows we propose the exact steps we need to take to implement PSS. This is a first iteration of Partial Set Security, and therefore we present the most minimal solution that achieves PSS possible.
+
+## Decision
+In Replicated Security, 95% (see  soft opt-out) of the top Cosmos Hub validators have to secure a consumer chain. In Partial Set Security (PSS) we introduce a parameter `N` that is associated with a consumer chain and require that the top `N%` of the Cosmos Hub validators have to secure this consumer chain. Additionally, we allow validators that do not belong to the top `N%` of the Cosmos Hub validators to dynamically opt in if they want to validate on the consumer chain or not.
+For example, if a consumer chain has `N = 95%`, then it ultimately receives the same security it receives today by Replicated Security. On the other hand, if a consumer chain has `N = 0`%, then no validator is forced to validate the chain but validators can opt in to do so instead.
+In what follows, we call a consumer chain _Top N_ if it has joined as Top N chain with `N > 0` and _Opt In_ chain otherwise.  An Opt In consumer chain is secured only by the validators that have opted in to secure that chain.
```

Comment:
@mpoke not sure what's your take on that (?)
We could only talk about `Top N` chains as you mentioned in the meeting but it seems to me that making the clear distinction between `Top N` and `Opt In` (i.e., `Top N` with `N = 0`) chains at the beginning simplifies the presentation because  later on we can just say "For Opt In chains we do this and for Top N chains we do that" instead of saying "If N = 0 then we do this and if N > 0 we do something else."